cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -146,19 +180,91 @@ public function is_media( $attachment_id ) {
 	}
 
 	/**
-	 * Get a resource type based on file.(Cloudinary v1 remove mime type in post data).
+	 * Convert media extension.
 	 *
-	 * @param \WP_Post|int $attachment_id The attachment ID or object.
+	 * @param string $filename The file to convert.
+	 *
+	 * @return string|null
+	 */
+	public function convert_media_extension( $filename ) {
+
+		// Add preferred formats in future.
+		$conversion_types = array(
+			'psd'	=> 'jpg',
+			'ai'	=> 'jpg',
+			'eps'	=> 'jpg',
+			'ps'	=> 'jpg',
+			'ept'	=> 'jpg',
+			'eps3'	=> 'jpg',
+			'indd'	=> 'jpg',
+			'webp'	=> 'gif',
+			'bmp'	=> 'jpg',
+			'fbx'	=> 'jpg',
+			'flif'	=> 'jpg',
+			'gltf'	=> 'jpg',
+			'heif'	=> 'jpg',
+			'heic'	=> 'jpg',
+			'ico'	=> 'png',
+			'svg'	=> 'png',
+			'tga'	=> 'jpg',
+			'tiff'	=> 'jpg',
+			'tif'	=> 'jpg'
+		);
+		$info             = pathinfo( $filename );
+		$extension        = strtolower( $info['extension'] );
+		$convert          = 'jpg'; // Default handler.
+
+		if ( ! empty( $conversion_types[ $extension ] ) ) {
+			$convert = $conversion_types[ $extension ];
+		}
+
+		$filename = trailingslashit( $info['dirname'] ) . $info['filename'] . '.' . $convert;
+
+		return $filename;
+	}
+
+	/**
+	 * Checks if the file is for preview only. True = only render sizes converted.
+	 *
+	 * @param int $attachment_id The attachment ID to check.
+	 *
+	 * @return bool
+	 */
+	public function is_preview_only( $attachment_id ) {
+		$preview_types = array(
```

Comment:
@DavidCramer 

I see value on a filter here as well, to allow injecting other previewable types.
What do you think?